docs/static/keystore.asciidoc

```diff
@@ -0,0 +1,152 @@
+[[keystore]]
+=== Secrets keystore
+
+When you configure Logstash, you might need to specify sensitive settings or
+configuration, such as passwords. Relying on the file system to protect these
+values is not sufficient. Logstash provides a keystore for storing secret values
+to use for configuration settings.
+
+Unlike the Elasticsearch keystore, the Logstash keystore does not store
+actual configuration settings. Instead, you add a key and secret value to the
+keystore, and then use the key in place of the secret value when you configure
+sensitive settings.
+
+The syntax for referencing keys is identical to the syntax for
+<<environment-variables, environment variables>>:
+
+`${KEY}`
+
+Where KEY is the name of the key.
+
+For example, imagine that the keystore contains a key called `ES_PWD` with the
+value `yourelasticsearchpassword`:
+
+* In the configuration file, use `output{  elasticsearch {  ...  password => "${ES_PWD}" }}}`
+* In `logstash.yml`, use: `xpack.management.elasticsearch.password: ${ES_PWD}`
+
+NOTE: Referencing keystore data from `pipelines.yml` or the command line (`-e`)
+is not currently supported.
+
+When Logstash parses the settings (`logstash.yml`) or configuration
+(`/etc/logstash/conf.d/*.conf`), it resolves keys from the keystore before
+resolving environment variables.
+
+// TODO: add keystore-command to running-logstash-command-line.asciidoc
+// To create and manage keys, use the `keystore` command. See the
+// <<keystore-command,command reference>> for the full command syntax, including
+// optional flags.
+
+[float]
+[[keystore-password]]
+=== Keystore password
+
+You can protect access to the Logstash keystore by storing a password in an
+environment variable called `LOGSTASH_KEYSTORE_PASS`. If you create the Logstash
+keystore after setting this variable, the keystore will be password protected.
+This means that the environment variable needs to be accessible to the running
+instance of Logstash. This environment variable must also be correctly set for
+any users who need to issue keystore commands (add, list, remove, etc.).
+
+Using a keystore password is optional. The data will be encrypted even if you
+do not set a password. However, it is highly recommended to configure the
+keystore password and grant restrictive permissions to any files that may
+contain the environment variable value.
+
+For example:
+[source,sh]
+--------------------------------------------------
+set +o history
```

Comment:
This is neat. Should we add `set -o history` after the `export`?

Reply:
Yup, that is much better, thanks! Change on https://github.com/elastic/logstash/pull/9044/commits/5713de6f50e1cf6befbc3ffb93984e2607ad4000